File: PHP/One/Transforms/FillAttributesFromTrivia.php
<?php

namespace One\Transforms\FillAttributesFromTrivia;

use One\Ast\Types\SourceFile;
use One\Ast\Types\IMethodBase;
use One\Ast\Types\IHasAttributesAndTrivia;
use One\Ast\Types\Package;
use One\Ast\Types\IMethodBaseWithTrivia;
use One\Ast\Statements\ForeachStatement;
use One\Ast\Statements\ForStatement;
use One\Ast\Statements\IfStatement;
use One\Ast\Statements\Block;
use One\Ast\Statements\WhileStatement;
use One\Ast\Statements\DoStatement;
use One\AstTransformer\AstTransformer;
use One\Ast\Expressions\Expression;

class FillAttributesFromTrivia extends AstTransformer {
    function __construct() {
        parent::__construct("FillAttributesFromTrivia");
        
    }
    
    protected function visitAttributesAndTrivia($node) {
        $node->attributes = FillAttributesFromTrivia::processTrivia($node->leadingTrivia);
    }
    
    protected function visitExpression($expr) {
        return $expr;
    }
    
    static function processTrivia($trivia) {
        $result = Array();
        if ($trivia !== null && $trivia !== "") {
            $regex = new \OneLang\RegExp("(?:\\n|^)\\s*(?://|#|/\\*\\*?)\\s*@([A-Za-z0-9_.-]+) ?((?!\\n|\\*/|$).+)?");
            while (true) {
                $match = $regex->exec($trivia);
                if ($match === null)
                    break;
                if (array_key_exists($match[1], $result))
                    // @php $result[$match[1]] .= "\n" . $match[2];
                    // @python result[match[1]] += "\n" + match[2]
                    // @csharp result[match[1]] += "\n" + match[2];
                    // @java result.put(match[1], result.get(match[1]) + "\n" + match[2]);
                    $result[$match[1]] .= "\n" . $match[2];
                else
                    $result[$match[1]] = ($match[2] ?? "") === "" ? "true" : $match[2] ?? "";
            }
        }
        return $result;
    }
}

File: CSharp/One/Transforms/FillAttributesFromTrivia.cs
using One.Ast;
using One;
using System.Collections.Generic;

namespace One.Transforms
{
    public class FillAttributesFromTrivia : AstTransformer
    {
        public FillAttributesFromTrivia(): base("FillAttributesFromTrivia")
        {
            
        }
        
        protected override void visitAttributesAndTrivia(IHasAttributesAndTrivia node)
        {
            node.attributes = FillAttributesFromTrivia.processTrivia(node.leadingTrivia);
        }
        
        protected override Expression visitExpression(Expression expr)
        {
            return expr;
        }
        
        public static Dictionary<string, string> processTrivia(string trivia)
        {
            var result = new Dictionary<string, string> {};
            if (trivia != null && trivia != "") {
                var regex = new RegExp("(?:\\n|^)\\s*(?://|#|/\\*\\*?)\\s*@([A-Za-z0-9_.-]+) ?((?!\\n|\\*/|$).+)?");
                while (true) {
                    var match = regex.exec(trivia);
                    if (match == null)
                        break;
                    if (result.hasKey(match.get(1)))
                        // @php $result[$match[1]] .= "\n" . $match[2];
                        // @python result[match[1]] += "\n" + match[2]
                        // @csharp result[match[1]] += "\n" + match[2];
                        // @java result.put(match[1], result.get(match[1]) + "\n" + match[2]);
                        result[match[1]] += "\n" + match[2];
                    else
                        result.set(match.get(1), (match.get(2) ?? "") == "" ? "true" : match.get(2) ?? "");
                }
            }
            return result;
        }
    }
}